supabase/migrations/24_billing_historicals.sql

```diff
@@ -0,0 +1,87 @@
+
+-- Always use a transaction, y'all.
+begin;
+
+-- Historical record of tenant billing statements. This structure
+-- comes from the return value of `billing_report_202308`.
+create table billing_historicals (
+    tenant              catalog_tenant  not null,
+    billed_month        timestamptz     not null,
+    line_items          jsonb           not null,
```

Comment:
I don't have confidence that these metrics are 100% right and durable, and I think we want to avoid codifying them in the structure of this table. Instead let's keep it really lean and just have:

* tenant
* billed_month
* report jsonb <- the complete jsonb output of the billing report

i suspect this also streamlines the UI, which is then just working with billing report JSON objects -- sometimes from this table, and sometimes from calling the API directly.

Reply:
Yup sounds good, I was on the fence about whether not we should be extracting these fields or not. Changed to capturing and JSON, and yes agreed it'll make the UI work easier in the short term, though it will make having inconsistent data in here easier in the long term which is why I split it out in the first place